### PR TITLE
BVN Consent Screen

### PR DESCRIPTION
**Pull request by jumaallan**
Story: https://app.shortcut.com/smileid/story/xxx

## Summary

A few sentences/bullet points about the changes

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

### CODE REVIEW

File: lib/src/main/java/com/smileidentity/compose/consent/bvn/BvnConsentScreen.kt [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+import com.smileidentity.compose.components.BottomPinnedColumn
+import com.smileidentity.compose.preview.Preview
+import com.smileidentity.compose.preview.SmilePreviews
+import com.smileidentity.viewmodel.BvnConsentViewModel
+import com.smileidentity.viewmodel.viewModelFactory
+
+@Composable
+fun BvnConsentScreen(
+    partnerIcon: Painter,
+    bvnDeliveryOptionsList: List<BvnDeliveryOptions>,
```
Comment (vanshg) [inline lib/src/main/java/com/smileidentity/compose/consent/bvn/BvnConsentScreen.kt:57]: Should be an ImmutableList

File: lib/src/main/java/com/smileidentity/compose/consent/bvn/BvnConsentScreen.kt [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+                        )
+                    }
+                }
+            }
+        },
+        modifier = modifier,
+    )
+}
+
+data class BvnDeliveryOptions(
```
Comment (vanshg) [inline lib/src/main/java/com/smileidentity/compose/consent/bvn/BvnConsentScreen.kt:295]: mark as internal?

File: lib/src/main/java/com/smileidentity/compose/consent/bvn/BvnInputScreen.kt [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                style = MaterialTheme.typography.headlineMedium,
+                fontWeight = FontWeight.ExtraBold,
+                modifier = Modifier.weight(1f),
+            )
+            Icon(
+                imageVector = Icons.Default.Close,
+                contentDescription = null,
+                modifier = Modifier
+                    .testTag("bvn_input_screen_cancel")
+                    .clickable { cancelBvnVerification() },
```
Comment (vanshg) [inline lib/src/main/java/com/smileidentity/compose/consent/bvn/BvnInputScreen.kt:84]: nit, we can pass method reference (should be better performance, albeit very minimal)?
```suggestion
                    .clickable(::cancelBvnVerification),
```

File: lib/src/main/java/com/smileidentity/compose/consent/bvn/BvnInputScreen.kt [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+            )
+        }
+        Spacer(modifier = Modifier.height(16.dp))
+        Button(
+            onClick = {
+                viewModel.handleEvent(
+                    BvnConsentEvent.SubmitBVNMode(
+                        bvn = bvn,
+                        country = BvnCountry.NIGERIA.country,
+                        idType = CountryIdType.NIGERIA_BVN.idType,
```
Comment (vanshg) [inline lib/src/main/java/com/smileidentity/compose/consent/bvn/BvnInputScreen.kt:118]: Seems like these should be properties saved inside the viewmodel instead

File: lib/src/main/java/com/smileidentity/models/Bvn.kt [hunk 5]
```diff
@@ -0,0 +1,109 @@
+package com.smileidentity.models
+
+import com.smileidentity.SmileID
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+@JsonClass(generateAdapter = true)
+data class BvnToptRequest(
```
Comment (vanshg) [inline lib/src/main/java/com/smileidentity/models/Bvn.kt:8]: nit: typo

```suggestion
data class BvnTotpRequest(
```

File: lib/src/main/java/com/smileidentity/models/Bvn.kt [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    @Json(name = "partner_id")
+    val partnerId: String = SmileID.config.partnerId,
+    @Json(name = "signature")
+    val signature: String,
+    @Json(name = "timestamp")
+    val timestamp: String,
+)
+
+@JsonClass(generateAdapter = true)
+data class BvnToptResponse(
```
Comment (vanshg) [inline lib/src/main/java/com/smileidentity/models/Bvn.kt:24]: ditto

File: lib/src/main/java/com/smileidentity/models/Bvn.kt [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    @Json(name = "id_number")
+    val idNumber: String,
+    @Json(name = "id_type")
+    val idType: String,
+    @Json(name = "partner_id")
+    val partnerId: String = SmileID.config.partnerId,
+    @Json(name = "signature")
+    val signature: String,
+    @Json(name = "timestamp")
+    val timestamp: String,
```
Comment (vanshg) [inline lib/src/main/java/com/smileidentity/models/Bvn.kt:20]: For requests, we should use this definition in order to continue supporting API key based approach, if the partner is using that

```suggestion
    @Json(name = "timestamp") val timestamp: String = System.currentTimeMillis().toString(),
    @Json(name = "signature") val signature: String = calculateSignature(timestamp),

```

Comment (vanshg) [reply lib/src/main/java/com/smileidentity/models/Bvn.kt:20]: same thing on the other `Request` classes

**Comment by jumaallan** [comment]
Closing in favour of #204

**Pull request closed without merging**